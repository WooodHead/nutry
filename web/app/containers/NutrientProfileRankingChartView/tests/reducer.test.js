
import {
  rankingResults } from 'fixtures/foodprofile';
import {
    loadRankings,
    loadRankingsSuccess,
    loadRankingsFailure } from '../actions';

import nutrientProfileRankingChartViewReducer from '../reducer';

describe('Container [FoodProfile] - reducers', () => {
  let state;
  beforeEach(() => {
  });
  it('should return the initial state', () => {
    expect(nutrientProfileRankingChartViewReducer(undefined, {})).toMatchSnapshot();
  });
  it('should handle the loadRankings action correctly', () => {
    expect(nutrientProfileRankingChartViewReducer(state, loadRankings())).toMatchSnapshot();
  });
  it('should handle the loadRankingsSuccess action correctly', () => {
    expect(nutrientProfileRankingChartViewReducer(state, loadRankingsSuccess(rankingResults))).toMatchSnapshot();
  });
  it('should handle the loadRankingsFailure action correctly on empty', () => {
    const err = new Error('failed to retrieve profile');
    expect(nutrientProfileRankingChartViewReducer(state, loadRankingsFailure(err))).toMatchSnapshot();
  });
});
