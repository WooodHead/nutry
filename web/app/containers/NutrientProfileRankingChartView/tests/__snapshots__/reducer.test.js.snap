// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [FoodProfile] - reducers should handle the loadRankings action correctly 1`] = `
Immutable.Map {
  "loading": true,
  "rankingResults": Immutable.Map {},
}
`;

exports[`Container [FoodProfile] - reducers should handle the loadRankingsFailure action correctly on empty 1`] = `
Immutable.Map {
  "loading": true,
  "rankingResults": Immutable.Map {},
}
`;

exports[`Container [FoodProfile] - reducers should handle the loadRankingsSuccess action correctly 1`] = `
Immutable.Map {
  "loading": false,
  "rankingResults": Immutable.Map {
    "ALA": Array [],
    "ARG": Array [],
    "CA": Array [
      Immutable.Map {
        "name": "raw beef thymus",
        "id": "13337",
        "value": 7,
        "unit": "mg",
      },
      Immutable.Map {
        "name": "braised beef thymus",
        "id": "13338",
        "value": 10,
        "unit": "mg",
      },
      Immutable.Map {
        "name": "Raw veal thymus",
        "id": "17218",
        "value": 3,
        "unit": "mg",
      },
      Immutable.Map {
        "name": "braised veal thymus",
        "id": "17219",
        "value": 4,
        "unit": "mg",
      },
      Immutable.Map {
        "name": "Fresh thyme",
        "id": "02049",
        "value": 405,
        "unit": "mg",
      },
      Immutable.Map {
        "name": "Dried thyme",
        "id": "02042",
        "value": 1890,
        "unit": "mg",
      },
    ],
    "CHOCDF": Array [
      Immutable.Map {
        "name": "raw beef thymus",
        "id": "13337",
        "value": 0,
        "unit": "g",
      },
      Immutable.Map {
        "name": "braised beef thymus",
        "id": "13338",
        "value": 0,
        "unit": "g",
      },
      Immutable.Map {
        "name": "Raw veal thymus",
        "id": "17218",
        "value": 0,
        "unit": "g",
      },
      Immutable.Map {
        "name": "braised veal thymus",
        "id": "17219",
        "value": 0,
        "unit": "g",
      },
      Immutable.Map {
        "name": "Fresh thyme",
        "id": "02049",
        "value": 24.45,
        "unit": "g",
      },
      Immutable.Map {
        "name": "Dried thyme",
        "id": "02042",
        "value": 63.94,
        "unit": "g",
      },
    ],
    "VITD": Array [
      Immutable.Map {
        "name": "braised veal thymus",
        "id": "17219",
        "value": 0,
        "unit": "ug",
      },
      Immutable.Map {
        "name": "Fresh thyme",
        "id": "02049",
        "value": 0,
        "unit": "ug",
      },
      Immutable.Map {
        "name": "Dried thyme",
        "id": "02042",
        "value": 0,
        "unit": "ug",
      },
    ],
  },
}
`;

exports[`Container [FoodProfile] - reducers should return the initial state 1`] = `
Immutable.Map {
  "loading": true,
  "rankingResults": Immutable.Map {},
}
`;
